src/mam4xx/aero_model.hpp

```diff
@@ -0,0 +1,634 @@
+  !-----------------------------------------------------------------*/
+  // @param [in] temp   ! air temperature [K]
+  // @return     dynamic viscosity of air, unit [g/cm/s]
+  return 1.8325e-4 * (416.16 / (temp + 120.)) * haero::pow(temp / 296.16, 1.5);
+
+} // end air_dynamic_viscosity
+
+KOKKOS_INLINE_FUNCTION
+Real air_kinematic_viscosity(const Real temp, const Real rhoair) {
+  /*-----------------------------------------------------------------
+  ! Calculate kinematic viscosity of air, unit [cm^2/s]
+  !-----------------------------------------------------------------*/
+  // @param [in] temp     ! air temperature [K]
+  // @param [in] rhoair   ! air density [g/cm3]
+  // @return     vsc_dyn_atm  ! dynamic viscosity of air [g/cm/s]
+  return air_dynamic_viscosity(temp) / rhoair;
+
+} // air_kinematic_viscosity
+
+KOKKOS_INLINE_FUNCTION
+void calc_rain_drop_conc(const int nr, const Real rlo, const Real dr,
+                         const Real rhoair,
+                         const Real precip, //! in
+                         Real rrainsv[nrainsvmax], Real xnumrainsv[nrainsvmax],
+                         Real vfallrainsv[nrainsvmax]) {
+  /*-----------------------------------------------------------------
+  !   compute rain drop number concentrations, radius and falling velocity
+  !-----------------------------------------------------------------*/
+  // @param [in] nr           number of rain bins
+  // @param [in] rlo          lower limit of rain radius [cm]
+  // @param [in] dr           rain radius bin width [cm]
+  // @param [in] rhoair       air mass density [g/cm^3]
+  // @param [in] precip       precipitation [cm/s]
+
+  // @param [out] rrainsv(:)   rain radius in each bin [cm]
+  // @param [out] xnumrainsv(:)  rain number concentration in each
+  // @param [out] vfallrainsv(:) rain droplet falling bin [#/cm3]
+  // @param [out] velocity [cm/s] bin [#/cm3]
+
+  const Real zero = 0;
+  Real precipsum = zero;
+  // loop over  cloud bins
+  for (int ii = 0; ii < nr; ++ii) {
+    // rain radius in the bin [cm]
+    const Real rr = rlo + ii * dr;
+    rrainsv[ii] = rr;
+    xnumrainsv[ii] = haero::exp(-rr / 2.7e-2);
+    // rain diameter in the bin [cm]
+    const Real dd = 2. * rr;
+    Real vfallstp = zero;
+    // BAD CONSTANT
+    if (dd <= 0.007) {
+      vfallstp = 2.88e5 * dd * dd;
+    } else if (dd <= 0.025) {
+      vfallstp = 2.8008e4 * haero::pow(dd, 1.528);
+    } else if (dd <= 0.1) {
+      vfallstp = 4104.9 * haero::pow(dd, 1.008);
+    } else if (dd <= 0.25) {
+      vfallstp = 1812.1 * haero::pow(dd, 0.638);
+    } else {
+      vfallstp = 1069.8 * haero::pow(dd, 0.235);
+    }
+    // rain droplet falling speed [cm/s]
+    vfallrainsv[ii] = vfallstp * haero::sqrt(1.204e-3 / rhoair);
+    // sum of precipitation in all bins
+    precipsum += vfallrainsv[ii] * rr * rr * rr * xnumrainsv[ii];
+
+  } // ii
+    // 1.333333 is simplified 4/3 for sphere volume calculation
```

Comment:
If you add a space above this line, `clang-format` will probably do a better job of indenting this comment.